js/TeamTopDonor.js

```diff
@@ -0,0 +1,96 @@
+(function () {
+  var app = angular.module( 'top-donor', [] );
+
+  app.controller( 'topDonorCtrl', function( $scope, $http ) {
+    var queryStringVars = GLOBALS.getUrlVars();
+    var teamId = queryStringVars['id'];
+    var topX = queryStringVars['topX'];
+
+    var GetTopDonors = function(data) {
+      // Need to flatten into an map of name: value
+      var groupedData = {};
+      $.each(data, function(i, val) {
+        // val is an array of "donor objects": { "name": name, "value": value }
+        $.each(val, function(i, donor) {
+          var name = donor.name;
```

Comment:
nitpicky, but: these lines feel unnecessary to me since the vars are only used twice in the next lines